integration/user_agent_test.go

```diff
@@ -0,0 +1,217 @@
+package main
+
+import (
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"sync"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+func TestUserAgent(t *testing.T) {
+	suite.Run(t, &userAgentSuite{})
+}
+
+type userAgentSuite struct {
+	suite.Suite
+}
+
+// mockRegistryHandler implements a minimal Docker Registry V2 API that captures User-Agent headers
+type mockRegistryHandler struct {
+	mu         sync.Mutex
+	userAgents []string
+}
+
+func (h *mockRegistryHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	// Capture the User-Agent header
+	h.mu.Lock()
+	h.userAgents = append(h.userAgents, r.Header.Get("User-Agent"))
+	h.mu.Unlock()
+
+	// Implement minimal Docker Registry V2 API endpoints
+	switch {
+	case r.URL.Path == "/v2/":
+		// Registry version check endpoint
+		w.Header().Set("Docker-Distribution-API-Version", "registry/2.0")
+		w.WriteHeader(http.StatusOK)
+
+	case strings.Contains(r.URL.Path, "/tags/list"):
+		// Return tags list
+		tags := map[string]interface{}{
+			"name": "test/image",
+			"tags": []string{"latest"},
+		}
+		w.Header().Set("Content-Type", "application/json")
+		w.WriteHeader(http.StatusOK)
+		if err := json.NewEncoder(w).Encode(tags); err != nil {
+			panic(err)
+		}
+
+	case strings.HasSuffix(r.URL.Path, "/manifests/latest"):
+		// Return a minimal OCI manifest
+		manifest := map[string]interface{}{
+			"schemaVersion": 2,
+			"mediaType":     "application/vnd.oci.image.manifest.v1+json",
+			"config": map[string]interface{}{
+				"mediaType": "application/vnd.oci.image.config.v1+json",
+				"digest":    "sha256:44136fa355b3678a1146ad16f7e8649e94fb4fc21fe77e8310c060f61caaff8a",
+				"size":      2,
+			},
+			"layers": []map[string]interface{}{
+				{
+					"mediaType": "application/vnd.oci.image.layer.v1.tar+gzip",
+					"digest":    "sha256:e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855",
+					"size":      0,
+				},
+			},
+		}
+		w.Header().Set("Content-Type", "application/vnd.oci.image.manifest.v1+json")
+		w.Header().Set("Docker-Content-Digest", "sha256:44136fa355b3678a1146ad16f7e8649e94fb4fc21fe77e8310c060f61caaff8a")
+		w.WriteHeader(http.StatusOK)
+		if err := json.NewEncoder(w).Encode(manifest); err != nil {
+			panic(err)
+		}
+
+	case strings.Contains(r.URL.Path, "/blobs/"):
+		// Return empty blob
+		w.WriteHeader(http.StatusOK)
+		if _, err := w.Write([]byte("{}")); err != nil {
```

Comment:
Well that’s not an empty blob, the comment could be confusing. I guess this is ~necessary as a “config” read by `skopeo inspect`? If so, the comment should say so.


Alternatively (preferably?), use `skopeo inspect --raw`, then the `/tags/list` and `/blobs/…` endpoints can be deleted entirely.